packages/qdrant-loader-core/tests/llm/providers/test_bedrock.py

```
@@ -0,0 +1,333 @@
+    response_body = ("{" + f'"embeddings": [{json.dumps(vector)}]' + "}").encode("utf-8")
+    client = _FakeBedrockClient(response_body=response_body)
+    mod = _reload_bedrock_module(monkeypatch, client)
+
+    settings = _make_llm_settings()
+    provider = mod.BedrockProvider(settings)
+
+    with pytest.raises(import_module("qdrant_loader_core.llm.errors").InvalidRequestError):
+        await provider.embeddings().embed(["hello"] * (mod.BedrockEmbeddings.MAX_BATCH_SIZE + 1))
+
+
+def test_factory_returns_bedrock_provider(monkeypatch):
```

Ruff (0.15.13)

[warning] Unused function argument: `monkeypatch`

(ARG001)